lib/base/object-packer.cpp

```diff
@@ -240,8 +239,8 @@ static void PackAny(const Value& value, StringBuilder& builder)
  */
 String icinga::PackObject(const Value& value)
 {
-	StringBuilder builder;
+	std::string builder;
 	PackAny(value, builder);
 
-	return builder.ToString();
+	return std::move(builder);
```

Comment:
```
/home/jbrost/dev/icinga2/lib/base/object-packer.cpp: In function ‘icinga::String icinga::PackObject(const icinga::Value&)’:
/home/jbrost/dev/icinga2/lib/base/object-packer.cpp:245:18: warning: redundant move in return statement [-Wredundant-move]
  245 |  return std::move(builder);
      |         ~~~~~~~~~^~~~~~~~~
/home/jbrost/dev/icinga2/lib/base/object-packer.cpp:245:18: note: remove ‘std::move’ call
```

`std::move` is strange, I know.

Reply:
1. ISO C++ or just your compiler?
2. If ISO, do all compilers comply to this?

Reply:
Oh wait, the return also implicitly performs a type conversion here. I'd have to look this up and usually take compiler warnings as good advice.

Reply:
IMAO it's actually quite simple: return calls the return type constructor w/ the returned value. In this case it shall be std::string&& to avoid one malloc().